icubam/www/handlers/upload_csv.py

```diff
@@ -20,7 +20,17 @@ def get_current_user(self):
 
   def gen_path(self, f_name):
     time_str = time.strftime("%Y-%m-%d-%H:%M:%S")
-    return os.path.join(self.upload_path, f"{time_str}-{f_name}")
+    # Sanitize the name some:
+    f_name = os.path.split(f_name)[1]
+    f_name, f_ext = os.path.splitext(f_name)
+    # Clean out non-alphanumerics (need to split ext first or we lose the '.'):
+    f_name, f_ext = [
+      "".join(c
+              for c in f_name
+              if c.isalnum())[:64]
+      for f_name in [f_name, f_ext]
+    ]
+    return os.path.join(self.upload_path, f"{time_str}-{f_name}.{f_ext}")
```

Comment:
A few unit tests wouldn't hurt to ensure the added functionality is working as expected. Can't this be done with a regex, something like?
```py
f_name = re.sub(r'\W+', '', f_name)
```

BTW, for path manipulations [pathlib](https://docs.python.org/3/library/pathlib.html)  is quite nice.

Reply:
Ok, I think os.path is sufficient here but will deff. give pathlib a good look.